core/src/main/java/io/snyk/plugins/artifactory/exception/SnykAPIFailureException.java

```diff
@@ -0,0 +1,14 @@
+package io.snyk.plugins.artifactory.exception;
+
+import io.snyk.sdk.api.v1.SnykResult;
+import io.snyk.sdk.model.TestResult;
+
+public class SnykAPIFailureException extends RuntimeException {
+  public SnykAPIFailureException(SnykResult<TestResult> result) {
+    super("Snyk API request was not successful. (" + result.statusCode + ")");
```

Comment:
There is also an "userMessage" field in the JSON response sometimes - could point you to failures like "wrong org selected" etc. Could we capture it and show it here if present?

Reply:
That may cause complexity due to additional JSON parsing. Especially in a constructor. If we do add that I think it should be in a separate PR and part of `SnykResult` (where we already do JSON parsing for `200` responses).